### PR TITLE
[KMcompiler]added the unpack_seq_triton op for Deepseek V4

### PR DESCRIPTION
**Pull request by hellojack163**
### PR Category
Operator | OP Test | Benchmark 

### Type of Change
New Feature

### Description
This PR added unpack_seq_triton operator, which ported from vllm.v1.attention.ops.common.unpack_seq_triton. This operator unpacks a padded batched tensor [B, Lmax, ...] back into a flat concatenated tensor [N, ...] using per-batch sequence lengths, where N = sum(lengths).

Triton kernel _unpack_seq_triton_kernel with a 3D launch grid (B, ceil(Lmax / BLOCK_T), ceil(D / BLOCK_D)) (default BLOCK_T=BLOCK_D=64, 4 warps)

For each batch element, computes the cumulative input offset from lengths, reads valid (non-padded) rows from the packed tensor, and writes them to contiguous output

Supports multi-dimensional inputs (>3 dims) by internally flattening to (B, Lmax, -1) and reshaping back

### Issue

N/A

### Correctness

The correctness test cases for the unpack_seq_triton operator have been ported from vLLM. All tests pass when executed with:
pytest tests/test_unpack_seq.py -s
image

### Performance
Benchmark configuration:

- Operator: unpack_seq_triton
- Comparison target: vLLM unpack_seq_triton
- Mode: kerel
- Warmup: 1000
- Iterations: 100
- Test command: pytest benchmark/test_unpack_seq.py -s

dtype | [N, D] | B | vLLM Latency(ms) | FlagGems latency(ms) | vLLM / FlagGems
-- | -- | -- | -- | -- | --
torch.float16 | [128, 64] | 5 | 0.050944 | 0.051008 | 0.999
torch.float16 | [1024, 128] | 4 | 0.05328 | 0.053408 | 0.998
torch.float16 | [2048, 256] | 5 | 0.05392 | 0.052096 | 1.035
torch.float16 | [512, 512] | 4 | 0.049504 | 0.048992 | 1.01
torch.float16 | [2048, 64] | 8 | 0.052288 | 0.051936 | 1.007
torch.float16 | [256, 1024 | 4 | 0.052288 | 0.052448 | 0.997
torch.float8_e4m3f | [128, 64] | 5 | 0.0528 | 0.049568 | 1.065
torch.float8_e4m3f | [1024, 128] | 4 | 0.049472 | 0.050464 | 0.98
torch.float8_e4m3f | [512, 512] | 4 | 0.04992 | 0.051168 | 0.976

### CODE REVIEW

File: benchmark/test_unpack_seq.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+def _is_cuda_hopper():
+    if flag_gems.device != "cuda":
+        return False
+    if not torch.cuda.is_available():
+        return False
+    major, minor = torch.cuda.get_device_capability()
+    return major >= 9
+
+
+CUDA_HOPPER = _is_cuda_hopper()
```
Comment (tengqm) [inline benchmark/test_unpack_seq.py]: I don't think you are checking if the device is Hopper (H20), you are checking if the device capability is > 8.9. In the future, this test may become confusing.
So ... please be honestly straightforward about the testing point.

File: benchmark/test_unpack_seq.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+# =============================================================================
+
+
+def _vllm_unpack_seq(packed, lengths):
+    return vllm_unpack_seq(packed, lengths)
+
+
+def _vllm_unpack_seq_fp8(packed, lengths):
+    return vllm_unpack_seq(packed, lengths)
+
```
Comment (tengqm) [inline benchmark/test_unpack_seq.py]: Please clarify why these wrappers are needed ...

File: benchmark/test_unpack_seq.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        lengths = torch.tensor(lengths_list, dtype=torch.int32, device=device)
+        packed = torch.randn(B, Lmax, D, dtype=torch.float32, device=device) * 0.1
+        packed_fp8 = packed.to(FP8_DTYPE)
+        yield packed_fp8, lengths
+
+
+# =============================================================================
+# Benchmark tests
+# =============================================================================
+
```
Comment (tengqm) [inline benchmark/test_unpack_seq.py]: Beautiful comments like this are unnecessary in this context.
We are not creating/modifying a file with thousands of lines.

File: benchmark/test_unpack_seq.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+@pytest.mark.unpack_seq
+@pytest.mark.skipif(
+    not (HAS_VLLM and CUDA_HOPPER),
+    reason="requires vLLM and NVIDIA Hopper architecture for FP8",
+)
+def test_unpack_seq_fp8():
+    bench = UnpackSeqFP8Benchmark(
+        op_name="unpack_seq_fp8",
```
Comment (tengqm) [inline benchmark/test_unpack_seq.py]: This will fail the CI benchmark job...
The pytest mark used has to be 

1) registered in the `conf/operators.yaml` file for it to be recoginizable;
2) use a consistent identifier when specifying `op_name`, otherwise the CI expects data for `unpack_seq` while the benchmark writes data to `unpack_seq_fp8`, thus a disgusting scenario to debug.

File: src/flag_gems/fused/__init__.py [hunk 5]
```diff
@@ -102,6 +103,7 @@
     "reshape_and_cache_flash",
     "rwkv_ka_fusion",
     "rwkv_mm_sparsity",
+    "unpack_seq_triton",
```
Comment (tengqm) [inline src/flag_gems/fused/__init__.py]: Please note that we keep this list sorted, always.

File: src/flag_gems/fused/unpack_seq.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    tl.store(out_row_ptr, packed_vals, mask=valid_row[:, None] & d_mask)
+
+
+def unpack_seq_triton(
+    packed_tensor: torch.Tensor,
+    lengths: torch.Tensor,
+    block_t: int = 64,
+    block_d: int = 64,
+) -> torch.Tensor:
+    original_shape = packed_tensor.shape
```
Comment (tengqm) [inline src/flag_gems/fused/unpack_seq.py:60]: I was expecting that a fused operator will do a `logger.debug("GEMS UNPACK_SEQ_TRITON")` invocation for problem diagnosis later.

File: src/flag_gems/fused/__init__.py (hunk 5 above)
Comment (tengqm) [reply src/flag_gems/fused/__init__.py]: One more thing ...
I believe you wanted to register this operator as "unpack_seq` as indicated in the `pytest.mark` idioms in the test case.

File: benchmark/test_unpack_seq.py (hunk 1 above)
Comment (hellojack163) [reply benchmark/test_unpack_seq.py]: changed the function name form _is_cuda_hoppe to is_cuda_available

File: benchmark/test_unpack_seq.py (hunk 2 above)
Comment (hellojack163) [reply benchmark/test_unpack_seq.py]: delete the wrappers

File: benchmark/test_unpack_seq.py (hunk 3 above)
Comment (hellojack163) [reply benchmark/test_unpack_seq.py]: delete the comments

File: benchmark/test_unpack_seq.py (hunk 4 above)
Comment (hellojack163) [reply benchmark/test_unpack_seq.py]: Added registration information in the conf/operators.yaml.

File: src/flag_gems/fused/unpack_seq.py (hunk 6 above)
Comment (hellojack163) [reply src/flag_gems/fused/unpack_seq.py:60]: Added a statement logger.debug("GEMS UNPACK_SEQ_TRITON")

File: src/flag_gems/fused/__init__.py (hunk 5 above)
Comment (hellojack163) [reply src/flag_gems/fused/__init__.py]: Adjust position of unpack_seq_triton in alphabetical order
